src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorConfigurationService.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class RazorConfigurationService
+    {
+        private readonly ILanguageServer _server;
+        private readonly ILogger _logger;
+
+        public RazorConfigurationService(ILanguageServer languageServer, ILoggerFactory loggerFactory)
+        {
+            if (languageServer is null)
+            {
+                throw new ArgumentNullException(nameof(languageServer));
+            }
+
+            if (loggerFactory is null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            _server = languageServer;
+            _logger = loggerFactory.CreateLogger<RazorConfigurationService>();
+        }
+
+        public async Task<RazorLSPOptions> GetLatestOptions()
+        {
+            try
+            {
+                var request = new ConfigurationParams()
+                {
+                    Items = new[]
+                    {
+                        new ConfigurationItem()
+                        {
+                            Section = "razor"
+                        },
+                    }
+                };
+
+                var result = await _server.Client.SendRequest<ConfigurationParams, object[]>("workspace/configuration", request);
+                if (result == null || result.Length < 1)
+                {
+                    return null;
+                }
+
+                var jsonString = result[0].ToString();
+                var builder = new ConfigurationBuilder();
+                var stream = new MemoryStream(Encoding.UTF8.GetBytes(jsonString));
+                builder.AddJsonStream(stream);
+                var config = builder.Build();
+
+                var instance = new RazorLSPOptions();
+                instance.EnableFormatting = bool.Parse(config["format:enable"]);
```

Comment:
My instinct is to have a `LoadConfig` method on `RazorLSPOptions` with this logic in it. Also probably a good idea to harden this against missing keys so that if a client sends incomplete  or malformed responses we'll get better messages at least.

Reply:
I wanted to keep the options POCO type free from too much logic. Mainly because, the extraction of options from the client settings could be different for different clients. (VS and VS Code may have different names for the same setting). I want `RazorConfigurationService` to be the only one aware of those differences.

Reply:
@NTaylorMullen is that true about client giving us different config formats? If so that sounds like a nightmare, I feel like it would really behoove us to normalize the configs before sending them up to the server, or even better store them in all in the same format (which the server already understands).

Reply:
> @NTaylorMullen is that true about client giving us different config formats? If so that sounds like a nightmare, I feel like it would really behoove us to normalize the configs before sending them up to the server, or even better store them in all in the same format (which the server already understands).

Currently very true. There are talks going on on how to unify all config formats for LSP though. Might be a while till a concrete strategy is decided on though.